wp/lib/bap_wp/src/z3_utils.ml

```diff
@@ -175,13 +176,15 @@ let asserts_of_model (model_string : string) (sym_names : string list) : Sexp.t
           warning "Warning: %s not instantiated in Z3 query\n" varname;
           Sexp.List [Sexp.Atom "assert" ; Sexp.Atom "true"]
         end
-    | _ -> failwith "Error: Unexpected form in external smt model"
+    | _ -> failwith (sprintf "Error: Unexpected form in external smt model: %s\n"
+                     model_string)
   in
   let model_sexp = Sexp.of_string model_string in
   match model_sexp with
-  | Sexp.List (Sexp.Atom "model" :: t) ->
+  | Sexp.List (Sexp.Atom "model" :: t) | Sexp.List t ->
     List.map ~f:process_decl t
-  | _ -> failwith "Error: Unexpected form in external smt model"
+  | _ -> failwith (sprintf "Error: Unexpected form in external smt model: %s\n"
```

Comment:
Can we make this message different from the one on line 179?

Reply:
Also, I've been using the informal convention `failwith "function_name: error message"` I dunno if we want this though.